### PR TITLE
Revert "Revert "Fold opencensus into grpc_impl namespace""

### PR DESCRIPTION
**Pull request by karthikravis**
Reverts grpc/grpc#18392

### CODE REVIEW

File: include/grpcpp/opencensus.h [hunk 1]
```diff
@@ -19,29 +19,20 @@
...
-
-// Returns the tracing Span for the current RPC.
-::opencensus::trace::Span GetSpanFromServerContext(ServerContext* context);
+static inline void RegisterOpenCensusPlugin() {
+  ::grpc_impl::RegisterOpenCensusPlugin();
+}
+static inline void RegisterOpenCensusViewsForExport() {
+  ::grpc_impl::RegisterOpenCensusViewsForExport();
+}
+static inline ::opencensus::trace::Span GetSpanFromServerContext(
```
Comment (g-easy) [inline include/grpcpp/opencensus.h]: Optional: this line is relying on a transitive `#include "opencensus/trace/span.h"`

File: include/grpcpp/opencensus_impl.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#define GRPCPP_OPENCENSUS_IMPL_H
+
+#include "opencensus/trace/span.h"
+
+namespace grpc {
+
+class ServerContext;
+}
+namespace grpc_impl {
+// These symbols in this file will not be included in the binary unless
```
Comment (g-easy) [inline include/grpcpp/opencensus_impl.h]: Should these comments be in the non-`_impl` file?

File: include/grpcpp/opencensus.h (hunk 1 above)
Comment (karthikravis) [reply include/grpcpp/opencensus.h]: This change just moves around namespace. Any other changes should be a future PR imo.

File: include/grpcpp/opencensus_impl.h (hunk 2 above)
Comment (karthikravis) [reply include/grpcpp/opencensus_impl.h]: As I understand, aim is to keep .h file as just a placeholder and move everything including the comments to _impl file. @nicolasnoble  to confirm.

Comment (g-easy) [reply include/grpcpp/opencensus_impl.h]: I feel like the _impl file is an extra indirection for the reader but it's up to gRPC folks to decide. :)

Comment (nicolasnoble) [reply include/grpcpp/opencensus_impl.h]: Yeah, we'll try to polish this a bit later. Right now this is kind of the best solution to keep doxygen results pretty.

**Pull request merged**